app/src/main/java/com/example/babytracking/PickerSetup.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.babytracking
+
+import android.app.DatePickerDialog
+import android.app.TimePickerDialog
+import android.content.Context
+import android.widget.TextView
+import com.example.babytracking.TimeFormatter
+import kotlinx.android.synthetic.main.sleep_layout.*
+import java.util.*
+
+/*
+This object allows the creation of date and time pickers.
+*/
+object PickerSetup {
+
+     fun setupDatePicker(context: Context, date_TextView: TextView, cal: Calendar) {
+
+        date_TextView!!.text = "--/--/----"
+
+        val dateSetListener =
+            DatePickerDialog.OnDateSetListener { _, yearDate, monthOfYear, dayOfMonth ->
+                cal.set(Calendar.YEAR, yearDate)
+                cal.set(Calendar.MONTH, monthOfYear)
+                cal.set(Calendar.DAY_OF_MONTH, dayOfMonth)
```

Comment:
Do you really need to use a Calendar? Why not returning a timestamp? If you need to return multiple fields, feel free to create a data class that contains the return values that you need.